### ISSUE TITLE
Invalid syntax  if (even an escaped) \ is the last character in quotes

### ISSUE THREAD

**Issue by YannikSc** [sunng87/handlebars-rust#422]
When defining a string in a template, which contains a `\` as last character (even escaped as `"\\"`) the renderer fails with an `InvalidSyntax` error.

Not sure if this is an intended behaviour, but it definitly feels like a bug and is also not the case in other parsers (like handlebars.js). Sorry if there is already another topic describing this, I have'nt found it.

here is an example which reproduces the bug:

```rust
#[macro_use]
extern crate handlebars;

use std::collections::BTreeMap;
use handlebars::Handlebars;

handlebars_helper!(replace: |input: str, from: str, to: str| {
    input.replace(from, to)
});

// Error when having a \ as last character in a string
fn main() {
    let data = BTreeMap::<String, String>::new();
    let mut bars = Handlebars::new();
    bars.register_helper("replace", Box::new(replace));

    // Replace a / with a \

    // Works with "\\ " (if the character, before the quote is not a "\" )
    println!("{}", bars.render_template(r#"{{replace "some/path" "/" "\\ " }}"#, &data).unwrap()); // works
    println!("{}", bars.render_template(r#"{{replace "some/path" "/" "\\something else" }}"#, &data).unwrap()); // works

    // Fails otherwise
    println!("{}", bars.render_template(r#"{{replace "some/path" "/" "\\" }}"#, &data).unwrap()); // fails
    println!("{}", bars.render_template(r#"{{replace "some/path" "/" "something before \\" }}"#, &data).unwrap()); // fails
}
```

**Comment by sunng87** [comment]
These seems to be affected by how we deal with escape in json. It works with [the original javascript version](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%3Cp%3E%7B%7Bfirstname%7D%7D%20%7B%7Blastname%7D%7D%20%7B%7Bloud%20%5C%22yes%20%5C%5C%5C%22%7D%7D%3C%2Fp%3E%20%5Cn%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20firstname%3A%20%5C%22Yehuda%5C%22%2C%5Cn%20%20lastname%3A%20%5C%22Katz%5C%22%2C%5Cn%7D%5Cn%22%2C%22output%22%3A%22%3Cp%3EYehuda%20Katz%20YES%20%5C%5C%3C%2Fp%3E%20%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('loud'%2C%20function(string)%20%7B%5Cn%20%20%20%20return%20string.toUpperCase()%5Cn%20%7D)%3B%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D). Let me investigate.

**Comment by sunng87** [comment]
@YannikSc the fix is available in 3.5.4
